botbot_plugins/plugins/vote.py

```diff
@@ -0,0 +1,394 @@
+        options =\
+            filter(bool, [option_name.strip() for option_name
+                          in (options.split(options_sep) if options else [])])
+        options = options or default_options
+        # Abstaining should always be an option
+        options.append(ABSTAIN_OPTION)
+        options = list(set(options))
+
+        # NOTE: this will create a for and against in __abstain__,
+        # but for simplicity's sake we won't try and prevent this.
+        # All abstainers will be added to __abstain__.for.
+        initial_votes = {
+            option_name: {FOR_NAME: [], AGAINST_NAME: []}
+            for option_name in options
+        }
+
+        self.store("name", json.dumps(name))
+        self.store("options", json.dumps(options))
+        self.store("votes", json.dumps(initial_votes))
+        self.store("changes_since_countvote", json.dumps(False))
+
+        return (INFO_MESSAGES["voting_started_name"].format(name=name)
+                if name else INFO_MESSAGES["voting_started"])
+
+    def _depends_on_votestarted(throw_error=True):
+        def decorator(func):
+            @wraps(func)
+            def wrap(*args, **kwargs):
+                slf = args[0]
+                if slf.retrieve("votes"):
+                    return func(*args, **kwargs)
+                elif throw_error:
+                    return ERROR_MESSAGES["no_vote_started"]
+                return ""
+            return wrap
+        return decorator
+
+    @listens_to_command("endvote")
+    @_depends_on_votestarted()
+    def endvote(self, line, args):
+        name = json.loads(self.retrieve("name"))
+        changes_since_countvote = json.loads(
+            self.retrieve("changes_since_countvote"))
+        reply = (INFO_MESSAGES["voting_ended_name"].format(name=name)
+                 if name else INFO_MESSAGES["voting_ended"])
+        if changes_since_countvote:
+            reply += "\n" + self._print_votes()
+
+        self.delete("name")
+        self.delete("options")
+        self.delete("votes")
+        self.delete("changes_since_countvote")
+
+        return reply
+
+    @listens_to_command("countvotes")
+    @_depends_on_votestarted()
+    def countvotes(self, line, args):
+        self.store("changes_since_countvote", json.dumps(False))
+        return self._print_votes()
+
+    vote_regex_template = (
+        u"^((?P<vote>(" +
+        u"|".join([re.escape(str_) for str_
+                   in VOTE_SHORTCUT_TO_NAME.keys()]) +
+        u"\\\\){vote_modifier})(?P<option>.*))$")
+
+    @listens_to_regex_command("vote", (vote_regex_template
+                                       .format(vote_modifier="?")))
+    @_depends_on_votestarted()
+    def vote(self, line, vote, option):
+        try:
+            changes_made = self._add_vote(self._parse_vote(line.user, (vote, option)))
+            if changes_made:
+                self.store("changes_since_countvote", json.dumps(True))
+        except InvalidOptionError, e:
```

Comment:
This is old style exception handling.

Please move to `except Exception as e` style.

Reply:
Fixed.